sbi/inference/posteriors/mcmc_posterior.py

```diff
@@ -321,6 +334,7 @@ def sample(
                     thin=thin,  # type: ignore
                     warmup_steps=warmup_steps,  # type: ignore
                     vectorized=(method == "slice_np_vectorized"),
+                    interchangeable_chains=True,
```

Comment:
This is just true for `slice_np` sampling? Or should we expose it via the `kwargs`?

Reply:
`interchangeable_chains` corresponds to whether the batch of `x` provided the sampler is iid or not. Right now, we only support `sample_batched` with `slice_np_vectorized`. In the future, we might want to support doing this with one of the other mcmc samplers we have available, in which case we can expose `interchangeable_chains` in the `kwargs` - although I would suggest this can be a future PR?